packages/snowpack/src/index.js

```diff
@@ -0,0 +1,61 @@
+import { compareSignatures /*, isComponent*/ } from '@prefresh/utils';
+
+export default function preactRefreshPlugin(config, pluginOptions) {
+	return {
+		knownEntrypoints: ['@prefresh/core'],
+		async transform({ contents, urlPath, isDev }) {
+			if (
+				!isDev ||
+				!urlPath.endsWith('.js') ||
+				urlPath.includes('web_modules') ||
+				urlPath.includes('node_modules')
+			)
+				return;
+
+			return {
+				result: `
+          import '@prefresh/core';
+          import * as $OriginalModule$ from ${JSON.stringify(urlPath)};
+          let $CurrentModule$ = $OriginalModule$;
+
+          const compareSignaturesForPrefreshment = ${compareSignatures.toString()};
+
+          window.$RefreshSig$ = () => {
+            let status = 'begin';
+            let savedType;
+            return (type, key, forceReset, getCustomHooks) => {
+              if (!savedType) savedType = type;
+              status = self.__PREFRESH__.sign(type || savedType, key, forceReset, getCustomHooks, status);
+            };
+          };
+
+          window.$RefreshReg$ = (type, id) => {
+            self.__PREFRESH__.register(type, module.i + ' ' + id);
+          };
+
+          ${contents}
+
+          if (import.meta.hot) {
+            import.meta.hot.accept(({ module }) => {
+              try {
+                for (let i in module) {
+                  if (typeof module[i] === 'function') {
+                    if (i in $CurrentModule$) {
+                      // We could add a check here on i.name if it's a component.
+                      compareSignaturesForPrefreshment(
+                        $CurrentModule$[i],
+                        module[i]
+                      );
+                    }
+                  }
+                }
+                $CurrentModule$ = module;
+              } catch(e) {
+                window.location.reload();
```

Comment:
Would it help to replace this with an import.meta.hot.reject() call instead? Feels lame to force you to reload yourself.

Reply:
Hmm, recently I discovered [this one](https://webpack.js.org/api/hot-module-replacement/#invalidate) which to me felt likea good course of action.

Reply:
Nice, +1 for matching their terminology.